igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -94,31 +80,22 @@ class BuildController {
     @RequestMapping(value = '/builds/queue/{master}/{item}')
     Object getQueueLocation(@PathVariable String master, @PathVariable int item) {
         def buildService = getBuildService(master)
-        if (buildService.buildServiceProvider() == BuildServiceProvider.JENKINS) {
-            try {
-                return buildService.getQueuedItem(item)
-            } catch (RetrofitError e) {
-                if (e.response?.status == NOT_FOUND.value()) {
-                    throw new NotFoundException("Queued job '${item}' not found for master '${master}'.")
-                }
-                throw e
-            }
-        } else if (buildService.buildServiceProvider() == BuildServiceProvider.TRAVIS) {
-            return buildService.queuedBuild(item)
+        if (buildService instanceof JenkinsService) {
+            JenkinsService jenkinsService = (JenkinsService) buildService
+            return jenkinsService.queuedBuild(item);
+        } else if (buildService instanceof TravisService) {
+            TravisService travisService = (TravisService) buildService
+            return travisService.queuedBuild(item)
         }
+        return null
```

Comment:
Should we explicitly throw and exception here as opposed to returning null and letting the upstream (downstream?) service deal with it?

Reply:
Yes, that makes sense---will throw an exception here